web/src/engine/providers/BookmarkPlugin.ts

```diff
@@ -24,6 +24,7 @@ export class BookmarkPlugin extends MediaContainer<Bookmark> {
 
     private Deserialize(serialized: BookmarkSerialized): Bookmark {
         const parent = this.plugins.WebsitePlugins.find(plugin => plugin.Identifier === serialized.Media.ProviderID);
+        if (!parent) return undefined; //if plugin is not found, invalidate bookmark
```

Comment:
I suggest to introduce a dedicated MediaContainer for orphaned bookmarks which may have a localised label such as `Website Terminated (${identifier})` and some kind of warning icon ⚠️.
Than there is no need for special handling of `undefined` in various places, as the parent is a valid MediaContainer.

Reply:
So a MediaItem i would apply as Parent to the bookmark in case MangaPlugin is not found on Load()? Am i right?

Reply:
Tried to do something like that.  For sure it needs improvement but am i on the right track now ?